apps/veil/src/entities/leaderboard/api/use-lp-leaderboard.ts

```diff
@@ -0,0 +1,73 @@
+import { useQuery, UseQueryResult } from '@tanstack/react-query';
+import { Position } from '@penumbra-zone/protobuf/penumbra/core/component/dex/v1/dex_pb';
+import { apiPostFetch } from '@/shared/utils/api-fetch';
+import {
+  LpLeaderboardRequest,
+  LpLeaderboardApiResponse,
+  LpLeaderboardSortKey,
+  LpLeaderboardSortDirection,
+  LqtLp,
+  LpLeaderboard,
+  LpLeaderboardResponse,
+} from '@/entities/leaderboard/api/utils';
+import { penumbra } from '@/shared/const/penumbra';
+import { DexService } from '@penumbra-zone/protobuf';
+import { bech32mPositionId } from '@penumbra-zone/bech32m/plpid';
+
+export const BASE_LIMIT = 10;
+export const BASE_PAGE = 1;
+
+// get the position state for each lp reward
+async function enrichLpLeaderboards(data: LqtLp[]): Promise<LpLeaderboard[]> {
+  if (data.length === 0) {
+    return [];
+  }
+
+  const positionIds = data.map(lp => lp.positionId);
+  const positionsRes = await Array.fromAsync(
+    penumbra.service(DexService).liquidityPositionsById({ positionId: positionIds }),
+  );
+  const positions = positionsRes.map(r => r.data).filter(Boolean) as Position[];
+
+  return data.map((lp, index) => ({
+    ...lp,
+    positionIdString: bech32mPositionId(lp.positionId),
+    position: positions[index] as unknown as Position,
+  }));
+}
```

Comment:
we should avoid duplicating this

Reply:
We can get around to this in a subsequent refactor.

Reply:
When `use-lp-leaderboard.ts`, no longer depends on the DexService for position states, we can remove this and there will be no duplicated code.